app/src/lib/git/tag.ts

```diff
@@ -32,3 +35,55 @@ export async function getAllTags(
 
   return tags.stdout.split('\n').filter(s => s !== '')
 }
+
+/**
+ * Fetches the tags that will get pushed to the remote repository (it does a network request).
+ *
+ * @param repository  - The repository in which to check for unpushed tags
+ * @param account     - The account to use when authenticating with the remote
+ * @param remote      - The remote to check for unpushed tags
+ * @param branchName  - The branch that will be used on the push command
+ */
+export async function fetchTagsToPush(
+  repository: Repository,
+  account: IGitAccount | null,
+  remote: IRemote,
+  branchName: string
+): Promise<ReadonlyArray<string>> {
+  const networkArguments = await gitNetworkArguments(repository, account)
+
+  const args = [
+    ...networkArguments,
+    'push',
+    remote.name,
+    branchName,
+    '--follow-tags',
+    '--dry-run',
+    '--porcelain',
+  ]
+
+  const result = await git(args, repository.path, 'fetchTagsToPush', {
+    env: await envForRemoteOperation(account, remote.url),
+  })
+
+  const lines = result.stdout.split('\n')
+  let currentLine = 1
+  const unpushedTags = []
+
+  while (lines[currentLine] !== 'Done') {
```

Comment:
had to look this up for myself, could we record it in a comment?
```suggestion
  // the last line of this porcelain command is always 'Done'
  while (lines[currentLine] !== 'Done') {
```